src/invoke_training/training/shared/hf_dataset.py

```diff
@@ -0,0 +1,129 @@
+import os
+import random
+
+import datasets
+import numpy as np
+import torch
+from accelerate import Accelerator
+from torch.utils.data import DataLoader
+from torchvision import transforms
+from transformers import CLIPTokenizer
+
+from invoke_training.training.lora.lora_training_config import DatasetConfig
+
+
+def initialize_hf_dataloader(
+    config: DatasetConfig,
+    accelerator: Accelerator,
+    tokenizer: CLIPTokenizer,
+    batch_size: int,
+) -> DataLoader:
+    """This function was mostly copied from
+    https://github.com/huggingface/diffusers/blob/a74c995e7da9cae390ea0918c490f2551a217659/examples/text_to_image/train_text_to_image.py
+
+    TODO(ryand): Review this function, add tests, and tidy it up.
```

Comment:
I intend to tidy up dataset handling and write some tests in a future PR once I've thought through the range of dataset formats that we want to support.